### PR TITLE
Improve (de)serialization of Raft log

### PR DESCRIPTION
**Pull request by as51340**
Fixes (de)serialization of CoordinatorClusterState, used in snapshot creation of NuRaft.
Endpoint refactored

[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/coordination/coordinator_cluster_state.cpp [hunk 1]
```diff
@@ -99,64 +108,37 @@ auto CoordinatorClusterState::DoAction(TRaftLog log_entry, RaftLogAction log_act
...
-      case ReplicationRole::REPLICA:
-        return "replica";
-    }
-  };
 
-  auto const entry_to_string = [&role_to_string](auto const &entry) {
-    return fmt::format("{}_{}", entry.first, role_to_string(entry.second.role));
-  };
+  // .at(0) is hack to solve the problem with json serialization of map
+  auto const log = nlohmann::json{instances_}.at(0).dump();
```
Comment (antoniofilipovic) [inline src/coordination/coordinator_cluster_state.cpp:115]: I would change this to be without braces, we could have a bug if we move to GCC or a different Clangd version
https://json.nlohmann.me/home/faq/#brace-initialization-yields-arrays

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Really good work :muscle: I would change this brace initialization so it doesn't bite us later

File: src/coordination/coordinator_cluster_state.cpp (hunk 1 above)
Comment (as51340) [reply src/coordination/coordinator_cluster_state.cpp:115]: Changing it here https://github.com/memgraph/memgraph/pull/1788/files#

**Review by antoniofilipovic — APPROVED** [review]

**Comment by as51340** [comment]
@kgolubic Let's add release note for this into 2.16. 
Fixed (de)serialization of CoordinatorClusterState, used in snapshot creation of NuRaft.

**Pull request merged**